bulk/management/commands/data_quality.py

```diff
@@ -0,0 +1,184 @@
+            "latest_bill_created_date": latest_bill_created_date,
+            "bill_id_with_latest_action": bill_with_latest_action_id,
+            "latest_action_date": latest_action_date,
+            "latest_action_description": latest_action_description}
+        )
+    return total_bills_per_session
+
+
+def average_number_data(bills, chambers):
+    average_num_data = defaultdict(list)
+
+    for chamber in chambers:
+        chamber_name = chamber.name.lower()
+        total_sponsorships_per_bill = []
+        total_actions_per_bill = []
+        total_votes_per_bill = []
+
+        average_sponsors_per_bill = 0
+        average_actions_per_bill = 0
+        average_votes_per_bill = 0
+
+        for bill in bills.filter(from_organization=chamber):
+            total_sponsorships_per_bill.append(bill.sponsorships.count())
+            total_actions_per_bill.append(bill.actions.count())
+            total_votes_per_bill.append(bill.votes.count())
+
+        average_sponsors_per_bill = round(mean(total_sponsorships_per_bill))
+        average_actions_per_bill = round(mean(total_actions_per_bill))
+        average_votes_per_bill = round(mean(total_votes_per_bill))
+
+        average_num_data[chamber_name].append({
+            "chamber": chamber_name,
+            "average_sponsors_per_bill": average_sponsors_per_bill,
+            "average_actions_per_bill": average_actions_per_bill,
+            "average_votes_per_bill": average_votes_per_bill}
+        )
+    return average_num_data
+
+
+def no_sources(bills, chambers):
+    no_sources_data = defaultdict(list)
+    for chamber in chambers:
+        chamber_name = chamber.name.lower()
+        total_bills_no_sources = bills.filter(from_organization=chamber, sources=None).count()
+        total_votes_no_sources = bills.filter(from_organization=chamber, votes__sources=None).count()
+        no_sources_data[chamber_name].append({
+            "chamber": chamber_name,
+            "total_bills_no_sources": total_bills_no_sources,
+            "total_votes_no_sources": total_votes_no_sources}
+        )
+    return no_sources_data
+
+
+def bill_subjects(bills, chambers):
+    bill_subjects_data = defaultdict(list)
+    for chamber in chambers:
+        chamber_name = chamber.name.lower()
+        overall_number_of_subjects = bills.distinct("subject").values_list("subject", flat=True).count()
+        number_of_subjects = bills.filter(from_organization=chamber).distinct("subject").values_list("subject", flat=True).count()
+        number_of_bills_without_subjects = bills.filter(from_organization=chamber, subject=None).count()
+        bill_subjects_data[chamber_name].append({
+            "chamber": chamber_name,
+            "overall_number_of_subjects": overall_number_of_subjects,
+            "number_of_subjects": number_of_subjects,
+            "number_of_bills_without_subjects": number_of_bills_without_subjects}
+        )
+    return bill_subjects_data
+
```

Comment:
some more additions: 
- bills without versions
- votes without any voters
- votes (which do have voters, as to not include above category) but where yes/no count do not match actual voters
-

Reply:
Added all of the above.